rspace/src/main/scala/coop/rchain/rspace/ReportingRspace.scala

```diff
@@ -125,6 +106,15 @@ class ReportingRspace[F[_]: Concurrent: ContextShift: Log: Metrics: Span, C, P,
   implicit protected[this] override lazy val MetricsSource: Metrics.Source =
     Metrics.Source(RSpaceMetricsSource, "reporting")
 
+  /**
+    * in order to distinguish the system deploy(precharge and refund) in the a normal user deploy
+    * It might be more easily to analyse the report with data structure
+    * Seq(Seq[ReportingEvent](Precharge), Seq[ReportingEvent](userDeploy), Seq[ReportingEvent](Refund))
+    * It would be seperated by the softcheckpoint creation.
+    */
+  val report: Ref[F, Seq[Seq[ReportingEvent]]] = Ref.unsafe(Seq.empty[Seq[ReportingEvent]])
+  val softReport: Ref[F, Seq[ReportingEvent]]  = Ref.unsafe(Seq.empty[ReportingEvent])
```

Comment:
It's better to not use `Ref.unsafe` but instead `Ref.of`, create Ref instances in companion object and supply these instances to the case class.

Reply:
@nzpr We can keep this for now to minimize scope of changes for this PR.

Reply:
It seems to me that in the second commit I reduced the amount of code without changing the functionality. Is there any difference in code execution between these two commits?